Wagram Multiple Refactoring and a Fix  ʕ   •`ᴥ•´ʔ

Waguramu opened the pull request:
Changes:

- Refactored Interpretation
    + minor refactoring of the Analyzers
    + minor RoboyEmotion refactoring
- Refactored Linguistics objects (Linguistics, Entity, DetectedEntity, Term, Triple, Concept)
- Refactored Inference + ExpoStates: removed code duplicates
- Updated Ontology
- Refactored Memory package, Memory nodes, Ontology: the nodes strictly abide by ontology, neo4j types are encapsulated, safe look-up through Ontology enums -> for the future Ontology decoupling
- Fixed Input + IO incompatibility with updated Interpretation

Unit tests:
+ Interpretation

ʕ•̫͡•ʕ•̫͡•ʔ•̫͡•ʔ•̫͡•ʕ•̫͡•ʔ•̫͡•ʕ•̫͡•ʕ•̫͡•ʔ•̫͡•ʔ•̫͡•ʕ•̫͡•ʔ•̫͡•ʔ♡

Commit:
Adapted dialog to directly included parser.

Commit:
Updated parser submodule checkout.

Aevyz committed:
RoslessCalls

Created RosLess Calls Class

Aevyz committed:
Alteration of Memory

All commented out due to testing issues

Aevyz committed:
Memory Update

Aevyz committed:
Git Ignore

Aevyz committed:
Memory is now Rosless

See pull request for more information

Aevyz committed:
Revert Git Ignore to prior Devel state

Aevyz committed:
Added Deprecation

Aevyz committed:
Refractor: Changed Name to Neo4jMemoryOperations

Aevyz committed:
Configuration: Start Memory & Init

Run Memory Main in Main of Dialog System
Made Init Memory Clients optional
Add Config to Start Memory and to Run Client Init

Aevyz committed:
Merge pull request #40 from Roboy/memory-ref-jason-3

Configuration: Start Memory & Init

Commit:
Moved parser resources to separate subfolder.

Commit:
Merge branch 'devel' of https://github.com/Roboy/roboy_dialog into joseph-roboy-parser-api

Commit:
Adjusted SemanticParserAnalyzer to new Linguistics API

Commit:
Updated parser submodule checkout

Commit:
Added roboy_parser submodule to roboy_dialog pom.xml.

Commit:
Updated submodule checkout.

Commit:
Fixed memory submodule checkout

Commit:
Updated dialog sumodule checkout.

Commit:
Fixed Parser dependency type.

Aevyz committed:
Comments and Code Quality Changes @nbasargin

Changes according to review

Aevyz committed:
More Code Quality Updates

Aevyz committed:
Updated POM for Jenkins

Aevyz committed:
Update POM for Jenkins

Aevyz committed:
Corrected POM File for use in Intellij

Intellij no longer breaks, I think...

Aevyz committed:
Updated Comments

Aevyz committed:
Removing Unneeded Code

Aevyz committed:
Memory Integration Tests --> No Update Test Yet

Update Test is currently broken

Aevyz committed:
Memory Integration Tests --> No Update Test Yet

Update Test is currently broken

Commit:
Merge branch 'memory-ref-jason-2' of https://github.com/Roboy/roboy_dialog into joseph-roboy-parser-api

Aevyz committed:
Documentation of New Class

Korn \m/

Aevyz committed:
Documentation of New Class

Korn \m/

Aevyz committed:
Integration Tests: Working Update Test

Bit of a hacky solution, but it will work for testing.

Aevyz committed:
Integration Tests: Working Update Test

Bit of a hacky solution, but it will work for testing.

Commit:
Fixed dialog multimodule layout

Commit:
Fixed parser runtime resource loading

Aevyz committed:
Test Name Consistency

Aevyz committed:
Test Name Consistency

Waguramu committed:
Input and IO related refactoring

Waguramu committed:
Linguistics refactoring

Waguramu committed:
Linguistics SentenceAnalysis + Interpretation refactoring

Waguramu committed:
RoboyEmotion refactoring

Waguramu committed:
Input object refactoring

Waguramu committed:
Inference refactoring

Waguramu committed:
ExpoStates refactoring : transition logic

Waguramu committed:
Ontology refactoring. Added enum methods for safe look-up

Waguramu committed:
Memory refactoring. MemoryNodes strictly abiding by the ontology

Waguramu committed:
Memory refactoring : encapsulated the type string for enum -> direct access by the enum values

Waguramu committed:
Updated Linguistics tests

Waguramu committed:
Merge branch 'joseph-roboy-parser-api' into wagram-ling-refactor

Waguramu committed:
Update ExpoIntroductionState.java

Fixed a bug: properties.get(<String>) -> properties.get(<Neo4jProperty>)

Waguramu committed:
Update MemoryNodeModel.java

Fixed a bug: <Neo4jRelationship> key -> <String> key.type

Aevyz committed:
Neo4j Update Actually Working

Considering my commit names, I'd fit right in with PHP development (see iso 8601 @ http://php.net/manual/en/class.datetime.

Aevyz committed:
Neo4j Update Actually Working

Considering my commit names, I'd fit right in with PHP development (see iso 8601 @ http://php.net/manual/en/class.datetime.

Aevyz committed:
Added/Updated Test Fail Messages

Aevyz committed:
Added/Updated Test Fail Messages

Aevyz committed:
Fixed Null Point Exception in Update Test

did not impact the test, but hey, the test looks cleaner ^^

Aevyz committed:
Fixed Null Point Exception in Update Test

did not impact the test, but hey, the test looks cleaner ^^

Aevyz committed:
Code Cleanup

Aevyz committed:
Code Cleanup

Aevyz committed:
Merge pull request #44 from Roboy/memory-integration-tests-jason

Merge Integration Tests into Main PR

Aevyz committed:
Partially standardized Error Msgs

Aevyz committed:
Partially standardized Error Msgs

missxa commented on src/main/java/roboy/dialog/states/expoStates/PersonalInformationAskingState.java:
```diff
@@ -53,9 +48,8 @@
-    private final String SELECTED_ABILITIES = "abilities";
-    private final String SELECTED_ROBOY_QA = "roboy";
+    private final String[] TRANSITION_NAMES = { "skills", "abilities", "roboy" };
+    private final String[] INTENT_NAMES = { "skills", "abilities", "roboy" };
```
i've seen these definitions in a few places, does it make sense to move them somewhere commonly accessible?

Waguramu replied:
Have to think. Maybe I'll ask @nbasargin about

missxa commented on src/main/java/roboy/dialog/states/expoStates/PersonalInformationAskingState.java:
```diff
@@ -142,11 +136,11 @@ public Output react(Interpretation input) {
 
         LOGGER.info(" -> Produced answer: " + answer);
-        nextState = getRandomTransition();
+        nextState = getRandomTransition(TRANSITION_NAMES, INTENT_NAMES, INTENTS_HISTORY_ID);
```
transitions should be passed in the state definition in JSON of a personality

missxa commented on src/main/java/roboy/dialog/states/expoStates/RoboyQAState.java:
```diff
@@ -83,18 +81,18 @@ public Output react(Interpretation input) {
                 }
             }
-            nextState = getRandomTransition();
+            nextState = getRandomTransition(TRANSITION_NAMES, INTENT_NAMES, INTENTS_HISTORY_ID);
```
same as above

missxa commented on src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java:
```diff
@@ -186,12 +155,12 @@ private String getRoboyFactsPhrase(Roboy roboy) {
+                HashMap<String, Integer> ages = new Agedater().determineAge(properties.get(birthdate).toString());
                 String retrievedAge = "0 days";
                 if (ages.get("years") > 0) {
                     retrievedAge = ages.get("years") + " years";
```
there's a bug apparently, cause Roboy is now saying he's 1 years old. either someone altered the birthdate in neo4j or there's an issue in the code

Waguramu replied:
Was a typo

missxa commented on src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java:
```diff
@@ -227,10 +196,10 @@ private String getRoboyFactsPhrase(Roboy roboy) {
             String nodeName = "";
-            if (node.getProperties().containsKey("full_name") && !node.getProperties().get("full_name").equals("")) {
-                nodeName = node.getProperties().get("full_name").toString();
+            if (node.getProperties().containsKey(full_name) && !node.getProperties().get(full_name).equals("")) {
```
would it be possible to have functions like:
- node.getProperties().getFullName() instead of node.getProperties().get(full_name)
- properties.getFuturePlans() instead of properties.get(future)
- etc.

Waguramu replied:
So, you propose to encapsulate? I think it makes sense to do for Roboy since we know what this is about, however, for a random MemoryNodeModel, I am not sure

missxa commented on src/main/java/roboy/linguistics/Concept.java:
```diff
@@ -1,5 +1,7 @@
 
+import java.util.Objects;
+
 public class Concept {
```
can you please add doxygen comments for this class

Waguramu replied:
Can try. But have no idea why we had this in the first place

missxa commented on src/main/java/roboy/linguistics/DetectedEntity.java:
```diff
@@ -1,5 +1,7 @@
 
+import java.util.Objects;
+
 public class DetectedEntity {
```
doxygen commentx

Waguramu replied:
Can try. But have no idea why we had this in the first place

missxa commented on src/main/java/roboy/linguistics/Entity.java:
```diff
@@ -2,6 +2,7 @@
 import java.util.Map;
+import java.util.Objects;
 
 public class Entity {
```
doxygen

Waguramu replied:
Can try. But have no idea why we had this in the first place

missxa commented on src/main/java/roboy/logic/Inference.java:
```diff
@@ -78,7 +82,44 @@ public String inferProperty(Neo4jProperty key, Interpretation input) {
-        return null;
+        // TODO we may need to do different inference
+        String result = null;
+        if (input.getSentenceType().compareTo(Linguistics.SentenceType.STATEMENT) == 0) {
```
rename compareTo-> equals

Waguramu replied:
Ok

missxa reviewed (approved):
does @Aevyz 's PR contain this changes already? if not, we should merge

missxa commented:
so minor changes to do, but generally good

josephbirkner commented:
I retargeted my dialog PR into this branch. It also now includes Jason's most recent changes.

josephbirkner committed:
Updated folder names and documentation.

Waguramu committed:
Various PR fixes:
- docstrings
- SemanticParserAnalyser extract_triples(<String>) fix
- MemoryNodeModel NullPointers fix
- Inference @Nullable calls fix
- Age output typo fix

Waguramu committed:
Merge branch 'wagram-ling-refactor' of github.com:Roboy/roboy_dialog into wagram-ling-refactor

Waguramu committed:
Need to sort out the resources

Waguramu committed:
Small fix <String> -> <Neo4jProperty>

Waguramu committed:
Merge branch 'joseph-roboy-parser-api' into wagram-ling-refactor

Waguramu committed:
Merged Joseph+Jason

Waguramu committed:
Updated .gitignore

Waguramu committed:
Merge branch 'memory-ref-jason-2' into wagram-ling-refactor

# Conflicts:
#	pom.xml
#	roboy_memory
#	src/main/java/roboy/memory/Neo4jMemoryOperations.java

Waguramu committed:
Removed RosMainNode from roboy.memory

Waguramu committed:
Merge branch 'memory-ref-jason-2' into wagram-ling-refactor

Waguramu committed:
Small fix for memory

missxa commented:
@Waguramu, so this includes @josephbirkner and @Aevyz changes; the Jenkins build is passing and the standard dialog flow is working?

Waguramu committed:
Disabled ROS by default
